contrib/coccinelle/no_hexdump.cocci

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium.
+/// Detect any usage of hexdump() or HEXDUMP() function calls, including
+/// suffixed variants. These should be avoided in production code.
+// Confidence: Medium
+// Options: --include-headers
+
+@initialize:python@
+@@
+
+import re
+cnt = 0
+
+@check_hexdump_call@
+expression args;
+identifier f;
+position p;
+@@
+
+f@p(args, ...);
+
+@script:python@
+p << check_hexdump_call.p;
+f << check_hexdump_call.f;
+@@
+
+if (
+    re.match(r"^(hexdump|HEXDUMP)", f)
+    and not p[0].file.endswith("lib/hexdump.h")
+    and "tests/" not in p[0].file
+):
+    print("* file %s: %s() used on line %s — avoid using hexdump/HEXDUMP calls in production" % (p[0].file, f, p[0].line))
+    cnt += 1
+
+@finalize:python@
+@@
+
+if cnt > 0:
+    print("""Remove hexdump/HEXDUMP usages from the code.
+""")
```

Comment:
This feels akward, but I was following the pattern in other files. At the same time, I think the print above exceeds the # of cols which might be the reason it was like this in first place

Reply:
:+1:

Reply:
Actioned